### ISSUE TITLE
Regroup the command implementations in respective sub-packages

### ISSUE THREAD

**Issue by navidshaikh** [knative/client#66]
Presently, all the command implementations reside under `pkg/kn/commands/`, we should re-group the implementations based on command-group.

for eg:

`pkg/kn/commands/service/`  <-- implementing all the sub-commands under `kn service`
`pkg/kn/commands/revision/`
`pkg/kn/commands/build/`
`pkg/kn/commands/`

and so on..

**Comment by maximilien** [comment]
So I took at pass at this when I was doing the `plugin` command. It's doable but requires the following changes:

1. move some types from `root.go` to a `common` package to avoid import cycles, e.g., this [common package](https://github.com/maximilien/client/blob/kn-plugins/pkg/kn/common) for one of my plugin branches

2. sweeping change to each command to use the common package above, e.g., this [plugins](https://github.com/maximilien/client/blob/kn-plugins/pkg/kn/commands/plugins/plugins.go) example

Bottom line is that it's not hard to do just requires agreement and one swoop change.

If you guys want this before milestone 1 I could do this this week, while you all a sleep in Europe. I'll add this to the agenda tomorrow as a low priority issue.

/cc @sixolet @cppforlife

**Comment by navidshaikh** [comment]
IMO, we should aim this as first thing after v0.1.

**Comment by rhuss** [comment]
@maximilien Is this something you are already on ? I'm asking while I'm working on #14 I also want to split up and separate the commands better (as I want to use some some package scoped vars which shouldn't 'pollute' whole "commands").

I agree that we need a "common" package which should hold the shared code, e.g. `KnParams`. I also believe that `KnParams` should be an interface defined in common and an implementation in the root "commands" package.

As I'm on it as well, I would be happy also to take over this task if you don't have a chance to come to is soonish. (there are some other subtle things like how to register subcommands and also the human-readable printers now reference commands directly).

In any case, we should tackle this issue very soon.

**Comment by maximilien** [comment]
OK as per discussion today. Let me submit something today. I am sure you guys will have lots of feedback. More soon.

**Comment by maximilien** [comment]
Done in #145

**Comment by navidshaikh** [comment]
This is completed, thanks @maximilien !
